### PR TITLE
Various small fixes

### PR DESCRIPTION
**Pull request by rhpvorderman**
This fixes issue #144 

Also I found various small issues while auditing the code. (After all the refactors, there were bound to be some loose ends). 

Since I did not want to make 10 different PRs each of these small changes are properly embedded in their own small commit. 

After this I think we are ready for 2.0.0!

### CODE REVIEW

**Comment by rhpvorderman** [comment]
@marcelm The requested reminder ping ;-).

**Review by marcelm — COMMENTED** [review]
Looks good to me! Feel free to merge with or without addressing the one comment I have.

File: src/xopen/__init__.py [hunk 1]
```diff
@@ -607,6 +631,9 @@ def _open_reproducible_gzip(filename, mode: str, compresslevel: int):
     # is called. This forces it to be closed.
     if closefd:
         gzip_file.myfileobj = fileobj
+    if sys.version_info.major == 3 and sys.version_info.minor < 12 and "r" not in mode:
```
[the remark this answers is not in the document]
Comment (rhpvorderman) [reply src/xopen/__init__.py:634]: That's much more readable! Thanks, I didn't know this.

**Comment by rhpvorderman** [comment]
Thanks for the review! I will prepare a 2.0.0 release.

**Pull request merged**